Assets/Tests/EditMode/AuthTests/HttpPortListenerTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Proyecto26;
+using Src.AuthController;
+using Src.AuthController.REST;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace Tests.EditMode.AuthTests
+{
+    public class HttpPortListenerTests
+    {
+        private HttpPortListener _listener;
+        private const int Port = 1234;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _listener = new HttpPortListener(Port);
+        }
+        
+        [Test]
+        public async Task StartListening_ShouldFireCallback()
+        {
+            string expectedCode = "someComplexCode";
+            string resultCode = "";
+            
+            void Callback(string code)
+            {
+                Debug.Log(code);
+                resultCode = code;
+
+                _listener.StopListening();
+            }
+
+            //UnityMainThreadDispatcher instancing
+            var go = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            var component = go.AddComponent<UnityMainThreadDispatcher>();
+         
+            RestClient.Request(new RequestHelper
```

Comment:
HttpProtListener should be designed so that it can be tested independently from other classes like RestClient.